BALSAMIC/utils/rule.py

```diff
@@ -350,3 +351,45 @@ def get_pon_samples(fastq_dir):
         (f.split("_1"))[0] for f in os.listdir(fastq_dir) if f.endswith("_R_1.fastq.gz")
     ]
     return samples
+
+
+def get_clinical_observations(config: dict) -> str:
+    """Returns path for clinical observations
+
+    Args:
+        congig: a config dictionary
+
+    Returns:
+        Path for clinical_snv_observations vcf file
+
+    """
+    if "clinical_snv_observations" in config["reference"]:
+        return Path(config["reference"]["clinical_snv_observations"]).as_posix()
```

Comment:
What should be returned in case there are no clinical observations? A `None`? an empty string? And how this empty field will be treated later by the rules?

Reply:
This method does not return `None`, so we have to provide a return statement for all of the possible cases.

So I would go for one of the following solutions, since we already have an `if` statement in `balsamic.smk`:

```suggestion
    return Path(config["reference"]["clinical_snv_observations"]).as_posix()
```
```suggestion
    if "clinical_snv_observations" in config["reference"]:
        return Path(config["reference"]["clinical_snv_observations"]).as_posix()
    return None
```